docs/semgrep-supply-chain/transitive-reachability.md

```diff
@@ -0,0 +1,116 @@
+
+
+:::info
+This feature is in private beta. To join, contact [support](/support).
+:::
+
+## Supported languages
+
+Semgrep currently performs transitive reachability analysis for JavaScript projects.
+
+## Definitions
+
+For this article:
+- **First-party code**: first-party code refers to your project, which includes the code that Semgrep scans
+- **Third-party code**: third-party code refers to the dependencies that the first-party code calls or imports and then uses
+
+Semgrep uses two overarching categories for project dependencies:
+
+- **Direct**: direct dependencies are those explicitly added by the developer, then called in the first-party code. You can visualize and review direct dependencies using the project's configuration file, such as the lockfile or the manifest file. Semgrep can also generate visual dependency graphs for projects that it scans.
+- **Transitive**: transitive dependencies are those that you include indirectly. For example, this happens when you directly add a dependency, but the dependency that you add then calls another dependency. In other words, a transitive dependency is one that third-party code adds and calls.
+
+## How transitive reachability analysis works
+
+Semgrep's reachability analysis determines whether there's a vulnerability in your codebase by checking your code, specifically the functions used and the function calls made. Then, Semgrep checks if the vulnerability is reachable by checking to see if:
+
+- The relevant function is called, which means that the code is used
+- The relevant function is called in a potentially unsafe way
+
+If neither condition is met, then Semgrep categorizes the vulnerability as **Unreachable**.
+
+Occasionally, Semgrep flags a finding as **needs review**. These findings are conditionally reachable, and you must manually review your code to validate whether the vulnerability is reachable or not.
+
+![A Supply Chain finding that needs review. Semgrep provides reachability details and remediation advice.](/img/vuln-needs-review.png#md-width)
+_**Figure**. A Supply Chain finding that needs review. Semgrep provides reachability details and remediation advice._
+
+The patterns that Semgrep Supply Chain uses to identify vulnerabilities present in first-party code are encapsulated in **rules**. With transitive reachability analysis, Semgrep extends its reachability analysis to the dependencies of dependencies to see if this code calls and uses vulnerable packages in a vulnerable way.
+
+To do this, Semgrep uses its Dependency Path feature to determine the set of packages, a subset of the third-party code, that calls on vulnerable packages. Semgrep then downloads the source code for the third-party dependencies called by your first-party code for analysis. <!-- TBD on whether we add a new CLI flag to control this behavior:`--allow-package-manager-install-deps` -->
```

Comment:
For lines 50-52:

The user might wonder "downloads to where?"

Maybe:

> Semgrep then downloads a copy of the third-party dependencies to its servers for scanning. It scans the third-party code using the same rules it uses against the first party code.

Or something similar?